cartridges/app_custom/cartridge/controllers/UserDetails.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+var server = require("server");
+var URLUtils = require("dw/web/URLUtils");
+var csrfProtection = require("*/cartridge/scripts/middleware/csrf");
+var userLoggedIn = require("*/cartridge/scripts/middleware/userLoggedIn");
+
+server.get(
+  "Add",
+  server.middleware.https,
+  csrfProtection.generateToken,
+  userLoggedIn.validateLoggedIn,
+  function (req, res, next) {
+    var accountHelpers = require("*/cartridge/scripts/account/accountHelpers");
+    var accountModel = accountHelpers.getAccountModel(req);
+
+    var actionUrl = URLUtils.url("UserDetails-SaveDetails");
+    var userDetailsForm = server.forms.getForm("userDetails");
+    userDetailsForm.clear();
+
+    userDetailsForm.firstname.value = accountModel.profile.firstName;
+    userDetailsForm.lastname.value = accountModel.profile.lastName;
+    userDetailsForm.phone.value = accountModel.profile.phone;
+    userDetailsForm.username.value = accountModel.profile.username;
```

Comment:
username is custom attribute which is not part of the accountModel, please add it to account model